package/MDAnalysis/lib/_cutil.pyx

```diff
@@ -65,6 +78,188 @@ def unique_int_1d(np.ndarray[np.int64_t, ndim=1] values):
+
+        # MDAnalysis can split molecules into their fragments
+        # based on bonding information.
+        # Note that this function will only handle a single fragment
+        # at a time, necessitating a loop.
+        for frag in u.fragments:
+          make_whole(frag)
+
+    Alternatively, to keep a single atom in place as the anchor::
+
+        # This will mean that atomgroup[10] will NOT get moved,
+        # and all other atoms will move (if necessary).
+        make_whole(atomgroup, reference_atom=atomgroup[10])
+
+
+    .. versionadded:: 0.11.0
+    """
+    cdef intset agset, refpoints, todo, done
+    cdef int i, nloops, ref, atom, other
+    cdef intmap bonding
+    cdef int[:, :] bonds
+    cdef float[:, :] oldpos, newpos
+    cdef bint ortho
+    cdef float[:] box
+    cdef float vec[3]
+
+    if reference_atom is None:
+        ref = atomgroup[0].index
+    else:
+        # Sanity check
+        if not reference_atom in atomgroup:
+            raise ValueError("Reference atom not in atomgroup")
+        ref = reference_atom.index
+
+    ortho = True
+    box = atomgroup.dimensions
+    for i in range(3):
+        if box[i + 3] != 90.0:
+            ortho = False
+
+    # set of indices in AtomGroup
+    agset = intset()
+    for i in atomgroup.indices.astype(np.int32):
+        agset.insert(i)
+    # C++ dict of bonds
+    try:
+        bonds = atomgroup.bonds.to_indices()
+    except (AttributeError, NoDataError):
+        raise NoDataError("The atomgroup is required to have bonds")
+    for i in range(bonds.shape[0]):
+        atom = bonds[i, 0]
+        other = bonds[i, 1]
+        # only add bonds if both atoms are in atoms set
+        if agset.count(atom) and agset.count(other):
+            bonding[atom].insert(other)
+            bonding[other].insert(atom)
+
+    oldpos = atomgroup.positions
+    newpos = np.zeros((oldpos.shape[0], 3), dtype=np.float32)
+
+    done = intset()  # Who have I already done?
+    refpoints = intset()  # Who is safe to use as reference point?
+    # initially we have one starting atom whose position we trust
+    refpoints.insert(ref)
+    for i in range(3):
+        newpos[ref, i] = oldpos[ref, i]
+
+    nloops = 0
+    while refpoints.size() < agset.size() and nloops < agset.size():
+        nloops += 1
+
+        # We want to iterate over atoms that are good to use as reference
+        # points, but haven't been done yet.
+        todo = difference(refpoints, done)
+        for atom in todo:
+            for other in bonding[atom]:
+                # If other is already a refpoint, leave alone
+                if refpoints.count(other):
+                    continue
+                # Draw vector from atom to other
+                for i in range(3):
+                    vec[i] = oldpos[other, i] - oldpos[atom, i]
+                # Apply periodic boundary conditions to this vector
+                if ortho:
+                    apply_pbc_ortho(&vec[0], &box[0])
```

Comment:
So here ideally we'd use the `minimum_image` things from calc_distances.h but there's type mismatches I need to figure out